File: 0274-h-index/0274-h-index.js
/**
 * @param {number[]} citations
 * @return {number}
 */
var hIndex = function(citations) {
    
    citations.sort((a,b)=>{
        if (a<b){
            return -1
        }
        else if (a>b){
            return 1
        }
        else{
            return 0
        }
    })
    
    n = citations.length
    for (let citation of citations){
        if (citation < n){
            n -= 1
        }
    }
    
    return n
};

File: 0274-h-index/0274-h-index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        
        for i in range(len(citations), 0, -1):
            
            val = 0
            for citation in citations:
                if citation >= i:
                    val += 1
                    
            if val >= i:
                return i
            
        return 0 # When h_index is not valid for researcher i.e., researchers doesn't have any published work
    
    
        
            
                
        